ndb/tests/conftest.py

```diff
@@ -23,16 +23,17 @@
 import pytest
 
 
-@pytest.fixture
-def property_clean_cache():
-    """Reset the ``_FIND_METHODS_CACHE`` class attribute on ``Property``
+@pytest.fixture(autouse=True)
+def reset_state():
+    """Reset module and class level runtime state.
 
-    This property is set at runtime (with calls to ``_find_methods()``), so
-    this fixture allows resetting the class to its original state.
+    To make sure that each test has the same starting conditions, we reset
+    module or class level datastructures that maintain runtime state.
+
+    This resets:
+
+    - ``model.Property._FIND_METHODS_CACHE``
+    - ``model.Model._kind_map``
     """
-    assert model.Property._FIND_METHODS_CACHE == {}
-    try:
-        yield
-    finally:
-        assert model.Property._FIND_METHODS_CACHE != {}
-        model.Property._FIND_METHODS_CACHE.clear()
+    model.Property._FIND_METHODS_CACHE.clear()
+    model.Model._kind_map.clear()
```

Comment:
Why this vs.

```python
assert model.Property._FIND_METHODS_CACHE == {}
assert model.Model._kind_map == {}
yield
model.Property._FIND_METHODS_CACHE.clear()
model.Model._kind_map.clear()
```

If you had that it would presumably have caught the fact that `ManyFields` modified `model.Model._kind_map` **outside** of a test.

Reply:
I simplified the reset fixture because if it's going to be run on every single test, we don't really need to concern ourselves with the state before it gets called--we just need to make the state what we want. The more complex checking made more sense when we were selectively applying the fixture, as it would let us know if there was a test that needed the fixture that hadn't gotten it. I didn't add a factory on the ManyFields class because none of the tests in which it's used rely on the `model.Model._kind_map`, so it doesn't make any practical difference.

Reply:
I worry that we'll later introduce a test that uses `ManyFields` and will assume `"ManyFields"` is a key in `model.Model._kind_map`. I'm advocating for the `== {}` approach (along with using a class factory for `ManyFields`) to mitigate against future situations where failures seem unrelated to the test code being added.

Reply:
Ok. 31b1548e6cb3d263805df483e29c0fcc2547376f